core/collateral/engine.go

```diff
@@ -3526,22 +3526,13 @@ func (e *Engine) CreatePartyAMMsSubAccounts(
 	ctx context.Context,
 	party, subAccount, asset, market string,
 ) (general *types.Account, margin *types.Account, err error) {
-	generalID := e.accountID(noMarket, subAccount, asset, types.AccountTypeGeneral)
-	if _, ok := e.accs[generalID]; ok {
-		return nil, nil, errors.New("general sub account already exists")
-	}
-
-	marginID := e.accountID(market, subAccount, asset, types.AccountTypeMargin)
-	if _, ok := e.accs[marginID]; ok {
-		return nil, nil, errors.New("general sub account already exists")
-	}
 
-	_, err = e.CreatePartyGeneralAccount(ctx, subAccount, asset)
+	generalID, err := e.CreatePartyGeneralAccount(ctx, subAccount, asset)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	_, err = e.CreatePartyMarginAccount(ctx, subAccount, market, asset)
+	marginID, err := e.CreatePartyMarginAccount(ctx, subAccount, market, asset)
 	if err != nil {
```

Comment:
Any chance that would be doing something weird in the case a party had created an amm, cancelled it, creating another one again?

Reply:
I have a test that does exactly that and its all fine. Both `CreatePartyGeneralAccount()` and `CreatePartyMarginAccount()` just return the ID if the account already exists, doesn't try to re-create or anything

Reply:
sounds good then, my bad I didn't catch it in the test.